File: 1064. 朋友数 (20).cpp
#include <stdio.h>
int main(){
    int N;
    scanf("%d",&N);
    int FriNum[N],FriSum[37] = {0}; //下标表示朋友证号
    for(int i=0;i<N;i++){
        scanf("%d",&FriNum[i]);
    }
    for(int i=0;i<N;i++){ //对每一个数字计数
        int sum = 0;
        while(FriNum[i]!=0){
            sum += FriNum[i] % 10;
            FriNum[i] = FriNum[i] / 10;
        }
        FriSum[sum] += 1;
    }
    int result[37] = {0},j=0;
    for(int i=0;i<37;i++)//查找非零项
        if(FriSum[i] != 0)
            result[j++] = i;
    printf("%d\n",j);
    printf("%d",result[0]);
    for(int i=1;i<j;i++)
        printf(" %d",result[i]);
}

File: 1064. 朋友数 (20).py
#输入部分
FriSum = [0 for i in range(37)] #下标表示朋友证号
N = eval(input(""))
FriNum = [int(i) for i in input("").split(" ")]

#对每一个数字计数
for i in range(len(FriNum)):
    sum = 0
    while FriNum[i]!=0:
        sum += FriNum[i] % 10
        FriNum[i] = FriNum[i] // 10
    FriSum[sum] += 1

查找非零项
result = []
for i in range(37):
    if FriSum[i] != 0:
        result.append(i)

#输出部分
print(len(result))
print(result[0],end='')
for i in range(1,len(result)):
    print(" %d"%result[i],end='')
